src/mangata/mangata.js

```diff
@@ -146,21 +153,28 @@ async function main() {
     const reservedDeltaBN = reservedPlanckDeltaBN.div(liquidityDecimalBN);
     console.log(`${account.name} has compounded ${reservedDeltaBN.toString()} more ${poolName} ...`);
 
-    console.log('\n5. Cancel task ...');
-    const cancelTaskExtrinsic = turingHelper.api.tx.automationTime.cancelTask(taskId);
-    await sendExtrinsic(turingHelper.api, cancelTaskExtrinsic, keyPair);
+    switch (actionSelected) {
+    case ScheduleActionType.executeOnTheHour: {
+        console.log('\n5. Cancel task ...');
+        const cancelTaskExtrinsic = turingHelper.api.tx.automationTime.cancelTask(taskId);
+        await sendExtrinsic(turingHelper.api, cancelTaskExtrinsic, keyPair);
 
-    const nextExecutionTime = executionTime + TASK_FREQUENCY;
-    const nextExecutionTimeout = calculateTimeout(nextExecutionTime);
+        const nextExecutionTime = executionTime + TASK_FREQUENCY;
+        const nextExecutionTimeout = calculateTimeout(nextExecutionTime);
 
-    console.log(`\n6. Keep Listening events on ${parachainName} until ${moment(nextExecutionTime * 1000).format('YYYY-MM-DD HH:mm:ss')}(${nextExecutionTime}) to verify that the task was successfully canceled ...`);
+        console.log(`\n6. Keep Listening events on ${parachainName} until ${moment(nextExecutionTime * 1000).format('YYYY-MM-DD HH:mm:ss')}(${nextExecutionTime}) to verify that the task was successfully canceled ...`);
 
-    const isTaskExecutedAgain = await listenEvents(mangataHelper.api, 'proxy', 'ProxyExecuted', nextExecutionTimeout);
-    if (isTaskExecutedAgain) {
-        console.log('Task cancellation failed! It executes again.');
-        return;
+        const isTaskExecutedAgain = await listenEvents(mangataHelper.api, 'proxy', 'ProxyExecuted', nextExecutionTimeout);
+        if (isTaskExecutedAgain) {
+            console.log('Task cancellation failed! It executes again.');
+            return;
+        }
+        console.log("Task canceled successfully! It didn't execute again.");
+        break;
+    }
+    default:
```

Comment:
Every case of switch has to be explictly listed out. You code is still hard for others to read. The right way to do this is,
```
const actionSelected = await askScheduleAction();

switch (actionSelected) {
    case ScheduleActionType.executeOnTheHour: {
          const schedule = Recurring;
          scheduleXcmpTask(schedule);
          cancelXcmpTask();
          break;
    }
    case ScheduleActionType.executeImmediately: // This can be the same with default but you have to declare it explicitly.
    default: {
          const schedule = { Fixed: { executionTimes: [0] } };
          scheduleXcmpTask(schedule);
          break;
    }
```
Please extract the logic of scheduleXcmpTask(schedule) first.